File: alt/01.cc
#include "iostream"
using namespace std;

int main() {
    int r1 = 0, i = 50;
    int r2 = 0, j = i;
    string line;
    while (cin >> line){
        char d = line[0];
        int n = stoi(line.substr(1));
        int lr = d == 'L' ? -1 : 1;
        i += n*lr;
        i %= 100;
        r1 += i==0;

        // p2
        int k = -1;
        while (++k < n){
            j += lr;
            j %= 100;
            r2 += j==0;
        }
    }
    cout << "res1/ " << r1 << endl;
    cout << "res2/ " << r2 << endl;
}

File: alt/01.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strconv"
)

func main(){

    r1 := 0
    r2 := 0
    i := 50
    j := i
    scanner := bufio.NewScanner(os.Stdin)
    for scanner.Scan(){
        line := scanner.Text()
        d := line[0]
        n, _ := strconv.Atoi(line[1:])
        lr := 1
        if d == 'L' {
            lr = -1
        }
        i += n*lr
        i %= 100
        if i == 0 {
            r1++
        }
        // p2
        for k:=0;k<n;k++ {
            j += lr
            j %= 100
            if j == 0 {
                r2++
            }
        }
    }
    fmt.Println("res1/", r1)
    fmt.Println("res2/", r2)
}

